app/controllers/actions/DataRequiredAction.scala

```diff
@@ -16,21 +16,26 @@
 
 package controllers.actions
 
-import javax.inject.Inject
 import models.UserAnswers
 import models.requests.{DataRequest, OptionalDataRequest}
 import play.api.mvc.{ActionRefiner, Result}
-//import play.api.mvc.Results.Redirect
-//import controllers.routes
+
+import java.time.Instant
+import javax.inject.Inject
 import scala.concurrent.{ExecutionContext, Future}
 
 class DataRequiredActionImpl @Inject()(implicit val executionContext: ExecutionContext) extends DataRequiredAction {
 
   override protected def refine[A](request: OptionalDataRequest[A]): Future[Either[Result, DataRequest[A]]] = {
 
     request.userAnswers match {
-      case None =>
-        Future.successful(Right(DataRequest(request.request, request.userId, UserAnswers(""), request.credId)))
+      case None => {
+        val userAnswers = request.userAnswers.getOrElse(UserAnswers(
```

Comment:
It feels a little pointless here doing `request.userAnswers.getOrElse` when we already know `request.userAnswers` is `None`? 

How about...

```
request.userAnswers match {
       case None =>
       val userAnswers = UserAnswers(
           id = request.userId,
           lastUpdated = Instant.now(java.time.Clock.systemUTC())
         )
         Future.successful(Right(DataRequest(request.request, request.userId, userAnswers, request.credId)))
         
         ...
```

Reply:
makes sense, changed